src/amo/components/Page/index.js

```diff
@@ -75,7 +77,9 @@ export const PageBase = ({
 
   return (
     <div className="Page-amo">
-      <VPNPromoBanner />
+      {showVPNPromo && clientApp !== CLIENT_APP_ANDROID ? (
```

Comment:
Could we check `clientApp` in `Addon`? That would put the entire "render or not render" logic in one location instead of two (I think `Addon` already has access to `clientApp`)

Reply:
Yeah, we could, and you're right `Addon` does already have access to `clientApp`. I agree having it all in one place would be nice, although, on the other hand, I can see how `Addon` could decide whether or not to show the promo based on the data of the add-on being displayed (in this case the `type` of add-on), but it makes a bit less sense for it to also care about `clientApp`, as that really has nothing to do with the add-on. That's why I thought it made more sense to have the `Page`, which is what ultimately decides whether to show the promo or not, be responsible for that logic.

Having said that, I think it makes sense to do what you suggest for now, but I wouldn't want us to add even more logic into `Addon` that isn't specific to add-ons later. For example, we're probably only going to want to show the promo to people in regions where it is supported. That will be dealt with via `withExperiment` for the experiment, but if we end up implementing something post-experiment that logic will have to live somewhere, and I'm not sure that place should be `Addon`.

Reply:
Right, this `clientApp` check should probably be moved in the "should exclude" function of the experiment.

Reply:
That's also the reason it was originally an `addonType` prop and not a `showVPNPromo` prop. Imo,  `Addon` shouldn't be responsible for deciding whether the promo is shown or not, it should only be responsible for giving `Page` the information it needs so that `Page` can make that decision. I guess having changed it to `showVPNPromo` we've already made the move towards `Addon` be responsible for making a decision, so your suggestion is just more of the same.

I'll make that change for the experiment, and we can revisit if we end up with something long-lived.

Reply:
Ah, I missed your last comment before adding mine. Sure, we could move it into the experiment logic for now.

Reply:
Actually, that's not going to work. :-(

The experiment configuration only takes effect when we enroll a user into the experiment. So if a user's first visit is to `/firefox/`, and they are enrolled in the experiment based on the fact that they visited a `/firefox/` page, but later they come back and visit an `/android/` page, they would still be considered to be enrolled and therefore they would still see the banner, even though we don't want to display the banner on `/android/` pages.

This is a bit of an edge case, but it does illustrate the fact that logic to enroll users into the experiment is not the same as logic used to display something on a page-by-page basis. Therefore I don't think it's the correct use of the "should exclude" function.

I'm going to land this without any logic for `clientApp` for now, as that seems the simplest approach, but we should add it somewhere other than in the "should exclude" logic for the experiment.

Reply:
Bummer.

I suppose we still need exclusion logic during enrollment and we cannot enroll everyone because only a subset of users will actually see this banner? We can probably share most of the logic between the exclusion function of the experiment and the conditional render in `Addon` or `Page`.

Reply:
I suppose we can exclude users who first land on an `/android/` page, because even if that means they will also never see the banner if they later visit a `/firefox/` page that's fine, as they're excluded and therefore we aren't paying attention to them. 

The idea of sharing the logic between the experiment and the display component is a good one.